Add quick start guide and new README (#1541)
kyma-project/api-gateway: docs/user/README.md

## What Is API Gateway?

> API Gateway is a Kyma module, which provides functionalities that allow you to expose and secure APIs.

Reviewer: ```suggestion
API Gateway is a Kyma module with which you can expose and secure APIs.
```

> To use the API Gateway module, the Istio module must be also added. By default, both the API Gateway and Istio modules are automatically added once you create a Kyma runtime instance. 

Reviewer: ```suggestion
To use the API Gateway module, you must also add the Istio module. By default, both the API Gateway and Istio modules are automatically added when you create a Kyma runtime instance. 
```

## Features

The API Gateway module offers the following features:

- Ory Oathkeeper installation: The module simplifies and manages the installation of Ory Oathkeeper.

> - API Exposure:  The module combines ORY Oathkeeper and Istio capabilities to offer the APIRule CustomResourceDefinition. By creating APIRule custom resources, you can easily and securely expose your workloads.

Reviewer: ```suggestion
- API Exposure: The module combines ORY Oathkeeper and Istio capabilities to offer the APIRule CustomResourceDefinition. By creating APIRule custom resources, you can easily and securely expose your workloads.
```

## What Is API Gateway?

API Gateway is a Kyma module, which provides functionalities that allow you to expose and secure APIs by using [Ory Oathkeeper](https://www.ory.sh/docs/oathkeeper) and [Istio](https://istio.io/) resources.

> By default, the API Gateway module is automatically added once you create a Kyma runtime instance. To use the API Gateway module, the Istio module must be also added.

Reviewer 2: You can mention that Istio, which API Gateway is dependant on is also enabled by default

Author: sure, great idea

## Features

The API Gateway module offers the following features:

- Ory Oathkeeper installation: The module simplifies and manages the installation of Ory Oathkeeper.
- API Exposure:  The module combines ORY Oathkeeper and Istio capabilities to offer the APIRule CustomResourceDefinition. By creating APIRule custom resources, you can easily and securely expose your workloads.
- Kyma Gateway installation: The module installs the default simple TLS Kyma Gateway known as Kyma Gateway.

## Architecture

![Kyma API Gateway Operator Overview](../assets/operator-overview.svg)

### API Gateway Operator

> Within the API Gateway moule, API Gateway Operator manages the application of API Gateway's configuration and handles resource reconciliation. It contains two controllers: [APIGateway Controller](./00-10-overview-api-gateway-controller.md), which applies the configuration specified in the [APIGateway custom resource (CR)](./custom-resources/apigateway/), and [APIRule Controller](./00-20-overview-api-rule-controller.md), which applies the configuration specified in the [APIRule CR](./custom-resources/apirule/).

Reviewer 2: ```suggestion
Within the API Gateway module, API Gateway Operator manages the application of API Gateway's configuration and handles resource reconciliation. It contains two controllers: [APIGateway Controller](./00-10-overview-api-gateway-controller.md), which applies the configuration specified in the [APIGateway custom resource (CR)](./custom-resources/apigateway/), and [APIRule Controller](./00-20-overview-api-rule-controller.md), which applies the configuration specified in the [APIRule CR](./custom-resources/apirule/).
```

### APIGateway Controller

APIGateway Controller manages the installation of [Ory Oathkeeper](https://www.ory.sh/docs/oathkeeper) and handles the configuration of Kyma Gateway and the resources defined in the APIGateway custom resource (CR). The controller is responsible for:
- Installing, upgrading, and uninstalling Ory Oathkeeper
- Configuring Kyma Gateway
- Managing Certificate and DNSEntry resources

> - Configuring Istio Gateways

Reviewer 2: please remove this point
